### PR TITLE
Static export macro for custom paths

### PR DESCRIPTION
**Pull request by welford**
See #1381 for original pull request.

### CODE REVIEW

**Comment by welford** [comment]
Added a macro for changing image paths also.

File: core/modules/widgets/image.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ ImageWidget.prototype.render = function(parent,nextSibling) {
 		tiddler = this.wiki.getTiddler(this.imageSource);
 	if(!tiddler) {
 		// The source isn't the title of a tiddler, so we'll assume it's a URL
-		src = this.imageSource;
+		src = this.getVariable("tv-get-export-image-link") || this.imageSource;
```
Comment (Jermolene) [inline core/modules/widgets/image.js]: I'd expect to be passing `this.imageSource` as a parameter to the `tv-get-export-image-link` macro:

```
src = this.getVariable("tv-get-export-image-link",
  {params: [{name: "title",value: this.imageSource}],
  defaultValue: this.imageSource}
);
```

And similarly for `tv-get-export-link` below.

**Comment by welford** [comment]
I have made the changes Jeremy suggested.

**Comment by Jermolene** [comment]
Thanks @welford sorry for the delay in merging.

File: core/modules/commands/rendertiddlers.js [hunk 2]
```diff
@@ -43,14 +44,19 @@ Command.prototype.execute = function() {
...
 		var container = $tw.fakeDocument.createElement("div");
 		widgetNode.render(container,null);
 		var text = type === "text/html" ? container.innerHTML : container.textContent;
-		fs.writeFileSync(path.resolve(pathname,encodeURIComponent(title) + extension),text,"utf8");
+		var export_path = null;
+		if($tw.utils.hop($tw.macros,"tv-get-export-path")) {
+			export_path = $tw.macros["tv-get-export-path"].run.apply(self,[title]);
+		}		
+		var final_path = path.resolve(outputPath,(export_path || encodeURIComponent(title)) + extension);
+		$tw.utils.createDirectory(path.dirname(final_path));
```
Comment (Jermolene) [inline core/modules/commands/rendertiddlers.js]: By the way, you could use `$tw.utils.createFileDirectories()` instead.

**Comment by Jermolene** [comment]
Hi @welford sorry for reverting this, my fault for merging without checking properly.

I found two issues:

* One test in `./bin/test.sh` fails; a link href that is expected to be `#This%20is%20my%20&#39;&#39;amazingly&#39;&#39;%20groovy%20macro!` is coming through as `This is my &#39;&#39;amazingly&#39;&#39; groovy macro!`
* rendertiddlers is saving tiddlers to the wrong directory; it seems to be ignoring the output path (I found the problem running `./quick-bld.sh` over in https://github.com/Jermolene/build.jermolene.github.io

File: core/modules/commands/rendertiddlers.js [hunk 3]
```diff
@@ -43,14 +44,19 @@ Command.prototype.execute = function() {
...
 			widgetNode = wiki.makeWidget(parser,{variables: {currentTiddler: title}});
 		var container = $tw.fakeDocument.createElement("div");
 		widgetNode.render(container,null);
 		var text = type === "text/html" ? container.innerHTML : container.textContent;
-		fs.writeFileSync(path.resolve(pathname,encodeURIComponent(title) + extension),text,"utf8");
+		var export_path = null;
+		if($tw.utils.hop($tw.macros,"tv-get-export-path")) {
+			export_path = $tw.macros["tv-get-export-path"].run.apply(self,[title]);
+		}		
+		var final_path = path.resolve(outputPath,(export_path || encodeURIComponent(title)) + extension);
```
Comment (Jermolene) [inline core/modules/commands/rendertiddlers.js]: Is the problem that here you should be referencing `pathname` instead of `outputPath`?

Comment (welford) [inline core/modules/commands/rendertiddlers.js]: In this case it is, but I have change it slightly in #1516

**Pull request merged**